src/ServiceControl.Transports.ASBS/QueueLengthProvider.cs

```diff
@@ -85,7 +85,20 @@ public Task Start()
                 while (await enumerator.MoveNextAsync().ConfigureAwait(false))
                 {
                     var queueRuntimeProperties = enumerator.Current;
-                    queuePathToRuntimeInfo.Add(queueRuntimeProperties.Name, queueRuntimeProperties);
+                    if (queuePathToRuntimeInfo.ContainsKey(queueRuntimeProperties.Name))
+                    {
+                        var existingItem = queuePathToRuntimeInfo[queueRuntimeProperties.Name];
+                        if(existingItem.ActiveMessageCount != queueRuntimeProperties.ActiveMessageCount)
+                        {
+                            Logger.WarnFormat(
+                                "Queue <{0}> already processed with different length (length {1} vs {2})",
+                                queueRuntimeProperties.Name,
+                                queuePathToRuntimeInfo.ActiveMessageCount,
+                                existingItem.ActiveMessageCount
+                                );
+                        }
+                    }
+                    queuePathToRuntimeInfo[queueRuntimeProperties.Name] = queueRuntimeProperties; // Assuming last write is most up to date
```

Comment:
```suggestion
                    if (queuePathToRuntimeInfo.ContainsKey(queueRuntimeProperties.Name))
                    {
                        Logger.WarnFormat("Queue <{0}> already processed.  Updating queue runtime properties.", queueRuntimeProperties.Name);
                    }
                    queuePathToRuntimeInfo[queueRuntimeProperties.Name] = queueRuntimeProperties; // Assuming last write is most up to date
```

So if we want to keep it simple, maybe something like this?